awslambdaric/lambda_runtime_client.py

```diff
@@ -68,10 +67,25 @@ def post_init_error(self, error_response_data):
         if response.code != http.HTTPStatus.ACCEPTED:
             raise LambdaRuntimeClientError(endpoint, response.code, response_body)
 
-    def wait_next_invocation(self):
-        with ThreadPoolExecutor() as e:
-            fut = e.submit(runtime_client.next)
-        response_body, headers = fut.result()
+    def wait_next_invocation(self, use_thread_for_polling_next=False):
+        # Calling runtime_client.next() from a separate thread unblocks the main thread,
+        # which can then process signals.
+        if use_thread_for_polling_next:
+            from concurrent.futures import ThreadPoolExecutor
```

Comment:
I think we can make it even better: let's move this import to `LambdaRuntimeClient::__init__` and make `use_thread_for_polling_next` as one of the argument of it (with the default value == False for backwards compatibility). Then, there would be no need to even add an extra argument to `wait_next_invocation`

Reply:
Moved `use_thread_for_polling_next ` under init, importing from init is not working.